sdk/provisioning/Azure.Provisioning/src/Infrastructure.cs

```diff
@@ -93,6 +94,97 @@ public virtual void Remove(Provisionable resource)
         }
     }
 
+    /// <summary>
+    /// Checks whether an name is a valid bicep identifier name comprised of
+    /// letters, digits, and underscores.
+    /// </summary>
+    /// <param name="identifierName">The proposed identifier name.</param>
+    /// <returns>Whether the name is a valid bicep identifier name.</returns>
+    public static bool IsValidIdentifierName(string? identifierName)
+    {
+        if (string.IsNullOrEmpty(identifierName)) { return false; }
+        if (char.IsDigit(identifierName![0])) { return false; }
+        foreach (char ch in identifierName)
+        {
+            if (!char.IsLetterOrDigit(ch) && ch != '_')
```

Comment:
Are we concerned about non-ascii letters?

Reply:
Bicep is underspecified, but a quick test fails for non-ASCII characters.  Tightening that up.